airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -392,17 +402,25 @@ public InputStream getErrorStream() {
     return this.stderr;
   }
 
+  /**
+   * Immediately terminates the Kube Pod backing this process and cleans up IO resources.
+   */
   @Override
   public int waitFor() throws InterruptedException {
     try {
       Pod refreshedPod = client.pods().inNamespace(podDefinition.getMetadata().getNamespace()).withName(podDefinition.getMetadata().getName()).get();
       client.resource(refreshedPod).waitUntilCondition(this::isTerminal, 10, TimeUnit.DAYS);
+      wasKilled.set(true);
       return exitValue();
     } finally {
       close();
     }
   }
 
+  /**
+   * Intended to gracefully clean up after a completed Kube Pod. This should only be called if the
+   * process is successful.
+   */
   @Override
```

Comment:
@jrhizor did I correctly interpret the difference between the two `waitFor` functions? I double checked the `WorkerUtils` class to confirm this is how they are used today.